{"ast":null,"code":"import _JSXStyle from \"styled-jsx/style\";\nimport React from \"react\";\nvar __jsx = React.createElement;\nimport Error from 'next/error';\nimport { STATIC_PATH } from '../constants/constants';\nimport * as Sentry from '@sentry/node';\nimport Color from 'color';\nimport { useContext } from 'react';\nimport { ThemeContext } from 'styled-components';\n\nfunction CustomError({\n  statusCode\n}) {\n  const theme = useContext(ThemeContext);\n  const backgroundColor = Color(theme.colors.background).hsl().string();\n  const title = statusCode === 503 ? 'Servidor no disponible' : null;\n  return __jsx(\"div\", {\n    className: _JSXStyle.dynamic([[\"2207119790\", [backgroundColor, STATIC_PATH]]]) + \" \" + \"error\"\n  }, __jsx(Error, {\n    statusCode: statusCode,\n    title: title\n  }), __jsx(_JSXStyle, {\n    id: \"2207119790\",\n    dynamic: [backgroundColor, STATIC_PATH]\n  }, [`.error>div{background:${backgroundColor} !important;color:#fff !important;}`, `.error>div>div::after{content:url(${STATIC_PATH}/logos/logo_project@2x.png);display:block;height:auto;margin-right:auto;margin-top:15px;margin-left:auto;max-width:100%;width:150px;}`, \".error>div h1{border-right-color:rgba(255,255,255,.3) !important;}\"]));\n}\n\nCustomError.getInitialProps = async ({\n  res,\n  err,\n  asPath\n}) => {\n  const errorInitialProps = await Error.getInitialProps({\n    res,\n    err\n  }); // Workaround for https://github.com/zeit/next.js/issues/8592, mark when\n  // getInitialProps has run\n\n  errorInitialProps.hasGetInitialPropsRun = true;\n\n  if (res) {\n    // Running on the server, the response object is available.\n    //\n    // Next.js will pass an err on the server if a page's `getInitialProps`\n    // threw or returned a Promise that rejected\n    Sentry.captureException(err);\n    return errorInitialProps; // Running on the client (browser).\n    //\n    // Next.js will provide an err if:\n    //\n    //  - a page's `getInitialProps` threw or returned a Promise that rejected\n    //  - an exception was thrown somewhere in the React lifecycle (render,\n    //    componentDidMount, etc) that was caught by Next.js's React Error\n    //    Boundary. Read more about what types of exceptions are caught by Error\n    //    Boundaries: https://reactjs.org/docs/error-boundaries.html\n  } // If this point is reached, getInitialProps was called without any\n  // information about what the error might be. This is unexpected and may\n  // indicate a bug introduced in Next.js, so record it in Sentry\n\n\n  Sentry.captureException(new Error(`_error.js getInitialProps missing data at path: ${asPath}`));\n  return errorInitialProps;\n};\n\nexport default CustomError;","map":{"version":3,"sources":["/Users/rafael/Projects/somosgad/dale/pages/_error.js"],"names":["Error","STATIC_PATH","Sentry","Color","useContext","ThemeContext","CustomError","statusCode","theme","backgroundColor","colors","background","hsl","string","title","getInitialProps","res","err","asPath","errorInitialProps","hasGetInitialPropsRun","captureException"],"mappings":";;;AAAA,OAAOA,KAAP,MAAkB,YAAlB;AACA,SAASC,WAAT,QAA4B,wBAA5B;AACA,OAAO,KAAKC,MAAZ,MAAwB,cAAxB;AACA,OAAOC,KAAP,MAAkB,OAAlB;AACA,SAASC,UAAT,QAA2B,OAA3B;AACA,SAASC,YAAT,QAA6B,mBAA7B;;AAEA,SAASC,WAAT,CAAqB;AAAEC,EAAAA;AAAF,CAArB,EAAqC;AACnC,QAAMC,KAAK,GAAGJ,UAAU,CAACC,YAAD,CAAxB;AACA,QAAMI,eAAe,GAAGN,KAAK,CAACK,KAAK,CAACE,MAAN,CAAaC,UAAd,CAAL,CAA+BC,GAA/B,GAAqCC,MAArC,EAAxB;AACA,QAAMC,KAAK,GAAGP,UAAU,KAAK,GAAf,GAAqB,wBAArB,GAAgD,IAA9D;AACA,SACE;AAAA,kDAOoBE,eAPpB,EAWqBR,WAXrB,aAAe;AAAf,KACE,MAAC,KAAD;AACE,IAAA,UAAU,EAAEM,UADd;AAEE,IAAA,KAAK,EAAEO;AAFT,IADF;AAAA;AAAA,cAOoBL,eAPpB,EAWqBR,WAXrB;AAAA,+BAOoBQ,eAPpB,4EAWqBR,WAXrB,gNADF;AA2BD;;AAEDK,WAAW,CAACS,eAAZ,GAA8B,OAAO;AAAEC,EAAAA,GAAF;AAAOC,EAAAA,GAAP;AAAYC,EAAAA;AAAZ,CAAP,KAAgC;AAC5D,QAAMC,iBAAiB,GAAG,MAAMnB,KAAK,CAACe,eAAN,CAAsB;AAAEC,IAAAA,GAAF;AAAOC,IAAAA;AAAP,GAAtB,CAAhC,CAD4D,CAG5D;AACA;;AACAE,EAAAA,iBAAiB,CAACC,qBAAlB,GAA0C,IAA1C;;AAEA,MAAIJ,GAAJ,EAAS;AACP;AACA;AACA;AACA;AAEAd,IAAAA,MAAM,CAACmB,gBAAP,CAAwBJ,GAAxB;AAEA,WAAOE,iBAAP,CARO,CASP;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACD,GAzB2D,CA2B5D;AACA;AACA;;;AACAjB,EAAAA,MAAM,CAACmB,gBAAP,CACE,IAAIrB,KAAJ,CAAW,mDAAkDkB,MAAO,EAApE,CADF;AAIA,SAAOC,iBAAP;AACD,CAnCD;;AAqCA,eAAeb,WAAf","sourcesContent":["import Error from 'next/error'\nimport { STATIC_PATH } from '../constants/constants'\nimport * as Sentry from '@sentry/node'\nimport Color from 'color'\nimport { useContext } from 'react'\nimport { ThemeContext } from 'styled-components'\n\nfunction CustomError({ statusCode }) {\n  const theme = useContext(ThemeContext)\n  const backgroundColor = Color(theme.colors.background).hsl().string()\n  const title = statusCode === 503 ? 'Servidor no disponible' : null\n  return (\n    <div className=\"error\">\n      <Error\n        statusCode={statusCode}\n        title={title}\n      />\n      <style global jsx>{`\n        .error > div {\n          background: ${backgroundColor} !important;\n          color: #fff !important;\n        }\n        .error > div > div::after {\n          content: url(${STATIC_PATH}/logos/logo_project@2x.png);\n          display: block;\n          height: auto;\n          margin-right: auto;\n          margin-top: 15px;\n          margin-left: auto;\n          max-width: 100%;\n          width: 150px;\n        }\n        .error > div h1 {\n          border-right-color: rgba(255, 255, 255, .3) !important;\n        }\n      `}</style>\n    </div>\n  )\n}\n\nCustomError.getInitialProps = async ({ res, err, asPath }) => {\n  const errorInitialProps = await Error.getInitialProps({ res, err })\n\n  // Workaround for https://github.com/zeit/next.js/issues/8592, mark when\n  // getInitialProps has run\n  errorInitialProps.hasGetInitialPropsRun = true\n\n  if (res) {\n    // Running on the server, the response object is available.\n    //\n    // Next.js will pass an err on the server if a page's `getInitialProps`\n    // threw or returned a Promise that rejected\n\n    Sentry.captureException(err)\n\n    return errorInitialProps\n    // Running on the client (browser).\n    //\n    // Next.js will provide an err if:\n    //\n    //  - a page's `getInitialProps` threw or returned a Promise that rejected\n    //  - an exception was thrown somewhere in the React lifecycle (render,\n    //    componentDidMount, etc) that was caught by Next.js's React Error\n    //    Boundary. Read more about what types of exceptions are caught by Error\n    //    Boundaries: https://reactjs.org/docs/error-boundaries.html\n  }\n\n  // If this point is reached, getInitialProps was called without any\n  // information about what the error might be. This is unexpected and may\n  // indicate a bug introduced in Next.js, so record it in Sentry\n  Sentry.captureException(\n    new Error(`_error.js getInitialProps missing data at path: ${asPath}`)\n  )\n\n  return errorInitialProps\n}\n\nexport default CustomError\n"]},"metadata":{},"sourceType":"module"}